2025-05-30 19:06:11,370 ERROR: Exception on / [GET] [in /home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask/app.py:1414]
Traceback (most recent call last):
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask/app.py", line 2190, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask/app.py", line 1486, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask_login/utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/app/main/routes.py", line 32, in index
    return render_template(
           ^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask/templating.py", line 151, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask/templating.py", line 132, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/home/edvin/Prog/spjald/app/templates/index.html", line 2, in top-level template code
    {% import 'bootstrap/wtf.html' as wtf %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/app/templates/base.html", line 1, in top-level template code
    {% extends 'bootstrap/base.html' %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 1, in top-level template code
    {% block doc -%}
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 4, in block 'doc'
    {%- block html %}
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 20, in block 'html'
    {% block body -%}
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 23, in block 'body'
    {% block content -%}
  File "/home/edvin/Prog/spjald/app/templates/base.html", line 54, in block 'content'
    {% block app_content %}{% endblock %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/app/templates/index.html", line 5, in block 'app_content'
    {% include '_post.html' %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/app/templates/_post.html", line 7, in top-level template code
    <a href="{{ url_for('main.user', username=post.author.username) }}">{{ post.author.name }} </a>
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/jinja2/environment.py", line 485, in getattr
    return getattr(obj, attribute)
           ^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'int object' has no attribute 'author'
2025-05-30 19:09:31,727 INFO: Spjald startup [in /home/edvin/Prog/spjald/app/__init__.py:86]
2025-05-30 20:01:15,244 INFO: Spjald startup [in /home/edvin/Prog/spjald/app/__init__.py:86]
2025-05-30 20:03:32,912 INFO: Spjald startup [in /home/edvin/Prog/spjald/app/__init__.py:86]
2025-05-30 20:03:39,494 ERROR: Exception on /search [GET] [in /home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask/app.py:1414]
Traceback (most recent call last):
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask/app.py", line 2190, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask/app.py", line 1486, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask_login/utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/app/main/routes.py", line 152, in search
    posts, total = Post.search(query_str, page, per_page)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/app/models.py", line 114, in search
    pag = q.paginate(page, per_page, False)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: Query.paginate() takes 1 positional argument but 4 were given
2025-05-30 20:05:40,012 INFO: Spjald startup [in /home/edvin/Prog/spjald/app/__init__.py:86]
2025-05-30 20:20:39,701 INFO: Spjald startup [in /home/edvin/Prog/spjald/app/__init__.py:86]
2025-05-30 20:20:40,767 ERROR: Exception on /login [GET] [in /home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask/app.py:1414]
Traceback (most recent call last):
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask/app.py", line 2190, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask/app.py", line 1486, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/app/auth/routes.py", line 26, in login
    return render_template('auth/login.html', title='Sign In', form=form)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask/templating.py", line 151, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask/templating.py", line 132, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/home/edvin/Prog/spjald/app/templates/auth/login.html", line 2, in top-level template code
    {% import 'bootstrap/wtf.html' as wtf %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/app/templates/base.html", line 1, in top-level template code
    {% extends 'bootstrap/base.html' %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 1, in top-level template code
    {% block doc -%}
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 4, in block 'doc'
    {%- block html %}
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 20, in block 'html'
    {% block body -%}
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 23, in block 'body'
    {% block content -%}
  File "/home/edvin/Prog/spjald/app/templates/base.html", line 54, in block 'content'
    {% block app_content %}{% endblock %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/app/templates/auth/login.html", line 17, in block 'app_content'
    <a href="{{ url_for('auth.reset_password_request') }}">Click to reset it</a>
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask/app.py", line 1697, in url_for
    return self.handle_url_build_error(error, endpoint, values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/flask/app.py", line 1686, in url_for
    rv = url_adapter.build(  # type: ignore[union-attr]
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/edvin/Prog/spjald/venv/lib/python3.12/site-packages/werkzeug/routing/map.py", line 950, in build
    raise BuildError(endpoint, values, method, self)
werkzeug.routing.exceptions.BuildError: Could not build url for endpoint 'auth.reset_password_request'. Did you mean 'auth.register' instead?
